tests/tools/fio/fio.go

```diff
@@ -5,45 +5,130 @@
 package fio
 
 import (
+	"fmt"
 	"io/ioutil"
 	"log"
+	"math/rand"
 	"os"
 	"os/exec"
+	"path"
+	"path/filepath"
 	"strings"
 	"sync"
+
+	"github.com/pkg/errors"
 )
 
 // List of fio flags
 const (
 	JobNameFlag = "--name"
 )
 
+const (
+	dockerExe            = "docker"
+	fioDataContainerPath = "/fio-data"
+)
+
+// Environment variable keys
+const (
+	// FioExeEnvKey gives the path to the fio executable to use in testing
+	FioExeEnvKey = "FIO_EXE"
+
+	// FioDockerImageEnvKey specifies the docker image tag to use. If
+	// FioExeEnvKey is set, the local executable will be used instead of
+	// docker, even if this variable is also set. The exception is if
+	// FioUseDockerEnvKey is not an empty string, which will force
+	// use of the fio docker image independent of FioExeEnvKey.
+	FioDockerImageEnvKey = "FIO_DOCKER_IMAGE"
+
+	// LocalFioDataPathEnvKey is the local path where fio data will be
+	// accessible. If not specified, defaults to the default temp directory (os.TempDir)
+	LocalFioDataPathEnvKey = "LOCAL_FIO_DATA_PATH"
+
+	// HostFioDataPathEnvKey specifies the path where fio data will be written,
+	// relative to the docker host. If left blank, defaults to local fio data path
+	// (works unless running via docker from within a container, e.g. for development)
+	HostFioDataPathEnvKey = "HOST_FIO_DATA_PATH"
+
+	// FioUseDockerEnvKey forces the fio runner to use the docker image, even if
```

Comment:
Is the "force" env var really needed? aren't `FIO_EXE` and `FIO_DOCKER_IMAGE` enough to control the desired behavior? or is there a particular case not handled with those?

Reply:
yeah it's not strictly needed. Was added as a convenience since setting and clearing this dumb switch was slightly easier than setting and clearing a path to the fio executable, plus it reduces ambiguity on what will be used if both EXE and DOCKER_IMAGE are set. I'll remove it.

Reply:
The precedence for which one is used should be documented. The ambiguity was present anyway when the "force" env var was not specified, or had a falsey value